staging/src/k8s.io/client-go/dynamic/simple.go

```diff
@@ -289,11 +289,9 @@ func (c *dynamicResourceClient) Get(ctx context.Context, name string, opts metav
 	if err != nil {
 		return nil, err
 	}
-	uncastObj, err := runtime.Decode(unstructured.UnstructuredJSONScheme, retBytes)
-	if err != nil {
-		return nil, err
-	}
-	return uncastObj.(*unstructured.Unstructured), nil
+	obj := &unstructured.Unstructured{}
+	err = runtime.DecodeInto(unstructured.UnstructuredJSONScheme, retBytes, obj)
+	return obj, err
```

Comment:
avoid returning obj in the error case:

```suggestion
	if err != nil {
		return nil, err
	}
	return obj, nil
```